src/control/cmd/daos_agent/mgmt_rpc_test.go

```diff
@@ -20,6 +22,80 @@ import (
 	"github.com/daos-stack/daos/src/control/logging"
 )
 
+func TestAgent_mgmtModule_getAttachInfo(t *testing.T) {
+	testResp := &mgmtpb.GetAttachInfoResp{
+		MsRanks: []uint32{0, 1, 3},
+		ClientNetHint: &mgmtpb.ClientNetHint{
+			Provider:    "ofi+sockets",
+			NetDevClass: netdetect.Ether,
+		},
+	}
+
+	testFI := &FabricInterface{
+		Name:        "test0",
+		Domain:      "",
+		NetDevClass: netdetect.Ether,
+	}
+
+	testRespWithHint := new(mgmtpb.GetAttachInfoResp)
+	*testRespWithHint = *testResp
+	testRespWithHint.ClientNetHint.Interface = testFI.Name
+	testRespWithHint.ClientNetHint.Domain = testFI.Name
+
+	for name, tc := range map[string]struct {
+		cacheDisabled bool
+		expErr        error
+		expResp       *mgmtpb.GetAttachInfoResp
+	}{
+		"cache disabled": {
+			cacheDisabled: true,
+			expResp:       testRespWithHint,
+		},
```

Comment:
This test did fail before I added the production code fix, since the original was trying to read back a cached value even if the cache was disabled. I can add some more to these test cases though.